### PR TITLE
Add CAVEATS file

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by chfast** [comment]
Please see the _Syntactic limits_ in the spec: https://webassembly.github.io/spec/core/appendix/implementation.html#syntactic-limits.

**Comment by axic** [comment]
I propose we also enforce some of the limits (such as the memory index in the data section) in the parser, because we are not creating a generic tool for manipulating binaries.

File: CAVEATS.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Caveats
```
Comment (chfast) [inline CAVEATS.md:1]: I'm not sure it is worth putting this in separate file.

Comment (axic) [reply CAVEATS.md:1]: Yeah I'm merging the README part into #278. The caveats file/contents is too early to be merged.

**Comment by axic** [comment]
This PR is not useful, moved to #282.

**Pull request closed without merging**